x-pack/filebeat/processors/aws_vpcflow/parse_aws_vpc_flow_log_test.go

```diff
@@ -0,0 +1,288 @@
+					"sublocation_type":    "wavelength",
+					"subnet_id":           "subnet-02d645xxxxxxxdbc0",
+					"tcp_flags":           int32(1),
+					"tcp_flags_array": []string{
+						"fin",
+					},
+					"traffic_path": int32(1),
+					"type":         "IPv4",
+					"version":      int32(5),
+					"vpc_id":       "vpc-09676f97xxxxxb8a7",
+				},
+			},
+			"cloud": mapstr.M{
+				"account": mapstr.M{
+					"id": "64111117617",
+				},
+				"availability_zone": "use1-az5",
+				"instance": mapstr.M{
+					"id": "i-0axxxxxx1ad77",
+				},
+				"region": "us-east-1",
+			},
+			"destination": mapstr.M{
+				"address": "10.200.0.0",
+				"ip":      "10.200.0.0",
+				"port":    int32(33004),
+			},
+			"event": mapstr.M{
+				"end":     end,
+				"outcome": "failure",
+				"start":   start,
+			},
+			"message": formatV5Sample,
+			"network": mapstr.M{
+				"bytes":       int64(1),
+				"direction":   "ingress",
+				"iana_number": "17",
+				"packets":     int64(52),
+				"protocol":    "udp",
+				"type":        "ipv4",
+			},
+			"related": mapstr.M{
+				"ip": []string{"89.160.20.156", "10.200.0.0", "10.200.0.80"},
+			},
+			"source": mapstr.M{
+				"address": "89.160.20.156",
+				"bytes":   int64(1),
+				"ip":      "89.160.20.156",
+				"packets": int64(52),
+				"port":    int32(50041),
+			},
+		}
+
+		assert.Equal(t, end, out.Timestamp)
+		if diff := cmp.Diff(expected, out.Fields); diff != "" {
+			t.Fatal(diff)
+		}
+	})
+}
+
+func TestGoldenFile(t *testing.T) {
+	testCases := readGoldenTestCase(t)
+
+	if *updateGolden {
+		// Delete existing golden files.
+		goldens, _ := filepath.Glob("testdata/*.golden.yml")
+		for _, golden := range goldens {
+			os.Remove(golden)
+		}
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+
+		t.Run(tc.Name, func(t *testing.T) {
+			c := defaultConfig()
+			c.Format = tc.Format
+			if tc.Mode != nil {
+				c.Mode = *tc.Mode
+			}
+
+			p, err := newParseAWSVPCFlowLog(c)
+			require.NoError(t, err)
+
+			observed := make([]mapstr.M, 0, len(tc.Samples))
+			for _, sample := range tc.Samples {
+				evt := &beat.Event{Fields: mapstr.M{"message": sample}}
+				out, err := p.Run(evt)
+				require.NoError(t, err)
+
+				if !out.Timestamp.IsZero() {
+					out.Fields["@timestamp"] = out.Timestamp
+				}
+				observed = append(observed, out.Fields)
+			}
+
+			goldenFile := filepath.Join("testdata", tc.Name+".golden.yml")
+			if *updateGolden {
+				writeGolden(t, goldenFile, observed)
+			} else {
+				expectedYAML := readGolden(t, goldenFile)
+
+				observedYAML, err := yaml.Marshal(observed)
+				require.NoError(t, err)
+
+				assert.YAMLEq(t, expectedYAML, string(observedYAML))
+			}
+		})
+	}
+}
+
+type GoldenTestCase struct {
+	Name    string   `yaml:"-"` // Name of test.
+	Mode    *mode    // Processing mode (what fields to generate).
+	Format  string   // Flow log format.
+	Samples []string // List of sample logs to parse.
+}
+
+func readGoldenTestCase(t testing.TB) []GoldenTestCase {
```

Comment:
Why `TB`?

Reply:
That's me following "accept interfaces, return structs" a little prematurely. Potentially I could run benchmarks of the different scenarios in testdata/aws-vpc-flow-logs.yml.

Reply:
I'd stay concrete until absolutely necessary.